src/wp-includes/comment-template.php

```diff
@@ -2011,9 +2018,15 @@ function comment_form_title( $no_reply_text = false, $reply_text = false, $link_
 	if ( 0 == $reply_to_id ) {
 		echo $no_reply_text;
 	} else {
+
 		// Sets the global so that template tags can be used in the comment form.
 		$comment = get_comment( $reply_to_id );
 
+		if ( 0 === (int) $comment->comment_approved ) {
```

Comment:
Should the post ID parameter be added to this so it can check the rendered and comment's post ID match? 

Actual question, not a passive aggressive change request ;)

Reply:
With the pending updates to the PR, if you try to pass `?replytocom=<comment ID on another post>`:

1. `comment_form_title()` will output `Leave a comment` if the comment ID being provided is for an **unapproved** comment on another post. In addition, the `comment_parent` field is correctly set to 0. This is the intended behaviour.

2. `comment_form_title()` will output `Reply to <parent comment author's name>` if the comment ID being provided is for an **approved** comment on another post. This is not the intended behaviour. However, the `comment_parent` field is correctly set to 0, so upon submission, the comment will be a top level comment on the current post - this part is intended behaviour.

Therefore, in order to prevent outputting the author of ** an approved comment** on **another post**, we _do_ need to add the `post_id` parameter to make sure they match. I'll add this to the PR update which I'll be pushing shortly.

Reply:
Thank you.

I'm on other tasks this week and vacation next week so I'll ask the tests team if they can do the branch testing. The code looks good though. 

cc @hellofromtonya